kraken-java-sdk/kraken-java-sdk-sync/src/main/java/com/consoleconnect/kraken/operator/sync/service/push/PushAPIActivityLogScheduler.java

```diff
@@ -0,0 +1,191 @@
+      return emptyList();
+    }
+  }
+
+  private List<PushExternalSystemPayload> pushLogs(MgmtEventEntity mgmtEvent) {
+    mgmtEvent.setStatus(EventStatusType.IN_PROGRESS.name());
+    mgmtEventRepository.save(mgmtEvent);
+    var logInfo = fromJson(mgmtEvent.getPayload(), PushLogActivityLogInfo.class);
+    try {
+      var sent = pushLogsInBatches(logInfo, mgmtEvent.getId());
+      mgmtEvent.setStatus(EventStatusType.DONE.name());
+      mgmtEventRepository.save(mgmtEvent);
+      return sent;
+    } catch (Exception ex) {
+      mgmtEvent.setStatus(EventStatusType.FAILED.name());
+      mgmtEventRepository.save(mgmtEvent);
+      return emptyList();
+    }
+  }
+
+  private List<PushExternalSystemPayload> pushLogsInBatches(
+      PushLogActivityLogInfo logInfo, UUID eventId) {
+    int page = 0;
+    var sent = new ArrayList<PushExternalSystemPayload>();
+    while (true) {
+      var pageable =
+          PageRequest.of(
+              page, getAppProperty().getControlPlane().getPushActivityLogExternal().getBatchSize());
+      var entities = getApiActivityLogRequestIds(logInfo, pageable);
+      var composedLogs = getComposedHttpRequests(entities.get());
+      if (composedLogs.isEmpty()) {
+        log.info("No API activity logs found to push to external system.");
+        break;
+      }
+      var payload =
+          new PushExternalSystemPayload(
+              eventId,
+              logInfo.getStartTime(),
+              logInfo.getEndTime(),
+              logInfo.getEnvName(),
+              toPageNoSubList(
+                  composedLogs,
+                  entities.getNumber(),
+                  entities.getSize(),
+                  entities.getTotalElements()));
+      var res = sendLogsToExternalSystem(payload);
+      if (res.getCode() != 200) {
+        throw new KrakenException(
+            400, "Pushing logs to external system filed with status: " + res.getCode());
+      }
+      page++;
+      sent.add(payload);
+    }
+    return sent;
+  }
+
+  private HttpResponse<String> sendLogsToExternalSystem(PushExternalSystemPayload payload) {
+    return blockCurl(
+        HttpMethod.POST, UriBuilder::build, payload, new ParameterizedTypeReference<>() {});
+  }
+
+  private List<ComposedHttpRequest> getComposedHttpRequests(Stream<ApiActivityLogEntity> logs) {
+    var requestIds = logs.map(AbstractHttpEntity::getRequestId).toList();
+    return this.apiActivityLogRepository.findAllByRequestIdIn(requestIds).stream()
+        .map(ApiActivityLogMapper.INSTANCE::map)
+        .collect(Collectors.groupingBy(ApiActivityLog::getRequestId))
+        .values()
+        .stream()
+        .map(
+            apiActivityLogs -> {
+              apiActivityLogs.sort(Comparator.comparing(ApiActivityLog::getCallSeq));
+              ComposedHttpRequest composedHttpRequest = new ComposedHttpRequest();
+              composedHttpRequest.setMain(apiActivityLogs.get(0));
+              if (apiActivityLogs.size() > 1) {
```

Comment:
Hi @jaroslawmalekcodete , what the purpose of this line and why does it set the sublist as branches?

Reply:
Hi @zhiyaoPCCW 
I used similar logic that is implemented in  ApiActivityLogService.java https://github.com/mycloudnexus/kraken/blob/main/kraken-java-sdk/kraken-java-sdk-core/src/main/java/com/consoleconnect/kraken/operator/core/service/ApiActivityLogService.java#L129 
The ComposedHttpRequest should allow to implemented Activity log push history in management plan.
Please let me know if we should change the response.